packages/amplify-graphql-sql-transformer/src/graphql-sql-transformer.ts

```diff
@@ -0,0 +1,148 @@
+    const env = context.synthParameters.amplifyEnvironmentName;
+
+    stack.templateOptions.templateFormatVersion = '2010-09-09';
+    stack.templateOptions.description = 'An auto-generated nested stack for the @sql directive.';
+
+    new cdk.CfnCondition(stack, ResourceConstants.CONDITIONS.HasEnvironmentParameter, {
+      expression: cdk.Fn.conditionNot(cdk.Fn.conditionEquals(env, ResourceConstants.NONE)),
+    });
+
+    this.sqlDirectiveFields.forEach((resolverFns) => {
+      resolverFns.forEach((config) => {
+        const { RDSLambdaDataSourceLogicalID: dataSourceId } = ResourceConstants.RESOURCES;
+        const dataSource = context.api.host.getDataSource(dataSourceId);
+        const MISSING_QUERY_STATEMENT = 'MISSING_CUSTOM_QUERY';
+        if (config.reference && !context.customQueries.has(config.reference)) {
+          throw new InvalidDirectiveError(
+            `@sql directive 'reference' argument must be a valid custom query name. Check type "${config.resolverTypeName}" and field "${config.resolverFieldName}". The custom query "${config.reference}" does not exist in "sql-statements" directory.`,
+          );
+        }
+
+        const statement = config.statement ?? (config.reference ? context.customQueries.get(config.reference) : MISSING_QUERY_STATEMENT);
+        const resolverResourceId = ResolverResourceIDs.ResolverResourceID(config.resolverTypeName, config.resolverFieldName);
+        const resolver = context.resolvers.generateQueryResolver(
+          config.resolverTypeName,
+          config.resolverFieldName,
+          resolverResourceId,
+          dataSource as any,
+          MappingTemplate.s3MappingTemplateFromString(
+            generateSqlLambdaRequestTemplate(statement ?? MISSING_QUERY_STATEMENT, 'RAW_SQL', config.resolverFieldName),
+            `${config.resolverTypeName}.${config.resolverFieldName}.req.vtl`,
+          ),
+          MappingTemplate.s3MappingTemplateFromString(
+            generateSqlLambdaResponseMappingTemplate(),
+            `${config.resolverTypeName}.${config.resolverFieldName}.res.vtl`,
+          ),
+        );
+
+        resolver.setScope(context.stackManager.getScopeFor(resolverResourceId, SQL_DIRECTIVE_STACK));
+        context.resolvers.addResolver(config.resolverTypeName, config.resolverFieldName, resolver);
+      });
+    });
+  };
+}
+
+export const generateSqlLambdaRequestTemplate = (statement: string, operation: string, operationName: string): string => {
+  return printBlock('Invoke RDS Lambda data source')(
+    compoundExpression([
+      set(ref('lambdaInput'), obj({})),
+      set(ref('lambdaInput.parameters'), obj({})),
+      set(ref('lambdaInput.statement'), str(statement)),
+      set(ref('lambdaInput.operation'), str(operation)),
+      set(ref('lambdaInput.operationName'), str(operationName)),
+      set(ref('lambdaInput.parameters'), methodCall(ref('util.defaultIfNull'), ref('context.arguments'), obj({}))),
+      obj({
```

Comment:
Is there a clean way to detect and fail fast if the variables expected by the SQL statement do not exist in the input? This will save customers invalid lambda invocations.